src/main/java/com/kustudents/issuetracker/service/ImportanceService.java

```diff
@@ -16,25 +16,56 @@
 
     private final ImportanceRepository importanceRepo;
 
+    public Importance createImportance(Importance importance){
+        return importanceRepo.save(importance);
+    }
+
+    public Importance createImportance(String importanceName){
+        Long importanceCount = importanceRepo.count();
+        Importance importance = new Importance();
+        importance.setName(importanceName);
+        importance.setSortOrder(importanceCount+1);
+        return importanceRepo.save(importance);
+    }
+
     public List<Importance> getImportanceList(){
-        return importanceRepo.findAll(Sort.by("id"));
+        return importanceRepo.findAll(Sort.by("sortOrder"));
     }
 
     public Importance getImportanceByID(Long ID){
-        return importanceRepo.getOne(ID);
+        return importanceRepo.findById(ID).orElseThrow();
     }
 
-    //TODO: Implement
-    public Page<Importance> getAllImportancesPaginatedAndFiltered(int page, int size, String orderBy, Boolean ascending){
-        return Page.empty();
+    public void rewriteImportances(List<Importance> importances){
+        importances.forEach((importance) ->{
+            updateImportance(importance);
+        });
     }
 
-    public void rewriteImportances(List<Importance> importances){
-        importanceRepo.deleteAllInBatch();
+    public void deleteImportanceById(Long id){
+        if(importanceRepo.count() == 1){
+            //TODO: Implement Exception
```

Comment:
throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "There can not be less than 1 importance"); - should work

Reply:
Implemented